<a href="https://colab.research.google.com/github/Pangilinancpe/CPE-031-CPE21S1---Visualizations-and-Data-Analysis---Pangilinan/blob/main/Assignment_1_1_Pangilinan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Assignment 1.1.

Introduction to Data Cleaning

## Objectives:


* Analyze and interpret the structure and content of raw datasets.

* Apply foundational data preprocessing techniques, such as handling missing values and data type conversions.

* Construct a basic data cleaning pipeline to process real-world data efficiently and with reproducibility.



## Procedures

Download the dataset for this activity [here](https://drive.google.com/file/d/1rNS4jxhZXN5LlLCmkoUW02UoT-UGUC46/view?usp=drive_link)

1. Load the dataset then show the first 5 and last 5 datapoints

In [ ]:
import pandas as pd

url = "https://drive.google.com/uc?export=download&id=1rNS4jxhZXN5LlLCmkoUW02UoT-UGUC46"

df = pd.read_csv(url)
df.head()

,Transaction ID,Item,Quantity,Price Per Unit,Total Spent,Payment Method,Location,Transaction Date
0,TXN_1961373,Coffee,2,2.0,4.0,Credit Card,Takeaway,2023-09-08
1,TXN_4977031,Cake,4,3.0,12.0,Cash,In-store,2023-05-16
2,TXN_4271903,Cookie,4,1.0,ERROR,Credit Card,In-store,2023-07-19
3,TXN_7034554,Salad,2,5.0,10.0,UNKNOWN,UNKNOWN,2023-04-27
4,TXN_3160411,Coffee,2,2.0,4.0,Digital Wallet,In-store,2023-06-11


In [ ]:
df.tail()

,Transaction ID,Item,Quantity,Price Per Unit,Total Spent,Payment Method,Location,Transaction Date
9995,TXN_7672686,Coffee,2,2.0,4.0,NaN,UNKNOWN,2023-08-30
9996,TXN_9659401,NaN,3,NaN,3.0,Digital Wallet,NaN,2023-06-02
9997,TXN_5255387,Coffee,4,2.0,8.0,Digital Wallet,NaN,2023-03-02
9998,TXN_7695629,Cookie,3,NaN,3.0,Digital Wallet,NaN,2023-12-02
9999,TXN_6170729,Sandwich,3,4.0,12.0,Cash,In-store,2023-11-07


QUESTION: What Column has numerical values?

the column in the dataset that has numerical values are, quantity, price per unit and the total spent

2. Check if there are any missing values in the dataset

In [ ]:
df.describe()

,Transaction ID,Item,Quantity,Price Per Unit,Total Spent,Payment Method,Location,Transaction Date
count,10000,9667,9862,9821,9827,7421,6735,9841
unique,10000,10,7,8,19,5,4,367
top,TXN_9226047,Juice,5,3.0,6.0,Digital Wallet,Takeaway,UNKNOWN
freq,1,1171,2013,2429,979,2291,3022,159


In [ ]:
df.isna().sum()

,0
Transaction ID,0
Item,333
Quantity,138
Price Per Unit,179
Total Spent,173
Payment Method,2579
Location,3265
Transaction Date,159


  QUESTION: How many columns had missing values? Which column had the highest number of missing values?

there are 7 total of columns that had a missing values, which are the items, quantity, price per unit, total spent, payment method, location, and transaction date. Those are the list of columns that have a missing values while the highest number of missing values in the list is the location, since it got 3,265 missing

3. Remove unneccessary columns and make sure numerical columns only contain numbers.

In [ ]:
df.columns

Index(['Transaction ID', 'Item', 'Quantity', 'Price Per Unit', 'Total Spent',
       'Payment Method', 'Location', 'Transaction Date'],
      dtype='object')

In [ ]:
df = df.drop(columns=['Transaction ID', 'Payment Method', 'Location', 'Transaction Date'])

df.head()

,Item,Quantity,Price Per Unit,Total Spent
0,Coffee,2,2.0,4.0
1,Cake,4,3.0,12.0
2,Cookie,4,1.0,ERROR
3,Salad,2,5.0,10.0
4,Coffee,2,2.0,4.0


In [ ]:
df.dtypes ## Checking the datatype for each column

,0
Item,object
Quantity,object
Price Per Unit,object
Total Spent,object


In [ ]:
df["Quantity"] = pd.to_numeric(df["Quantity"], errors='coerce')
df["Price Per Unit"] = pd.to_numeric(df["Price Per Unit"], errors='coerce')
df["Total Spent"] = pd.to_numeric(df["Total Spent"], errors='coerce')

df.head()

,Item,Quantity,Price Per Unit,Total Spent
0,Coffee,2.0,2.0,4.0
1,Cake,4.0,3.0,12.0
2,Cookie,4.0,1.0,NaN
3,Salad,2.0,5.0,10.0
4,Coffee,2.0,2.0,4.0


In [ ]:
df.dtypes ## Checking the datatype for each column

,0
Item,object
Quantity,float64
Price Per Unit,float64
Total Spent,float64


QUESTION: What happened to the object columns?




the columns originally typed as object are now converted into float64

4. Remove missing data to ensure all that the final dataset is clean

In [ ]:
df.isna().sum()

,0
Transaction ID,0
Item,333
Quantity,138
Price Per Unit,179
Total Spent,173
Payment Method,2579
Location,3265
Transaction Date,159


In [ ]:
df = df.dropna()

In [ ]:
df.isna().sum()

,0
Transaction ID,0
Item,0
Quantity,0
Price Per Unit,0
Total Spent,0
Payment Method,0
Location,0
Transaction Date,0


In [ ]:
df.head()

,Transaction ID,Item,Quantity,Price Per Unit,Total Spent,Payment Method,Location,Transaction Date
0,TXN_1961373,Coffee,2,2.0,4.0,Credit Card,Takeaway,2023-09-08
1,TXN_4977031,Cake,4,3.0,12.0,Cash,In-store,2023-05-16
2,TXN_4271903,Cookie,4,1.0,ERROR,Credit Card,In-store,2023-07-19
3,TXN_7034554,Salad,2,5.0,10.0,UNKNOWN,UNKNOWN,2023-04-27
4,TXN_3160411,Coffee,2,2.0,4.0,Digital Wallet,In-store,2023-06-11


EXPLAIN the process of data cleaning done in the procedure.

since the df.isna().sum() was called, it was to detect the missing values across the columns, then along df.dropna() to verify that there are no missing values remained before cleaning the dataset

## Supplemetary Activities

Download the dataset for supplementary activity [here](https://drive.google.com/file/d/1IGm1BrsLlud2QR0Z8_ix4aNw16YclO0E/view?usp=drive_link)



1.   Load the dataset and show the first 5 and last 5 datapoints
2.   Check if there are any missing values in the dataset.
3.   The df must only have Name, Subsribers, Category, and Country. Remove all other columns.
4.   Convert Subscribers to type **int**
5.   Ensure that there are no rows with missing values. (UNKNOWN is counted as a missing value)



In [ ]:
import pandas as pd

url = "https://drive.google.com/uc?export=download&id=1IGm1BrsLlud2QR0Z8_ix4aNw16YclO0E"

df = pd.read_csv(url)
df.head()

,Name,Brand channel,Subscribers (millions),Primary language,Category,Country
0,MrBeast,No,335,English,Entertainment,United States
1,T-Series,UNKNOWN,280,Hindi,Music,India
2,Cocomelon - Nursery Rhymes,Yes,UNKNOWN,UNKNOWN,Education,United States
3,SET India,Yes,UNKNOWN,Hindi,Entertainment,India
4,Vlad and Niki,No,129,English,Entertainment,Russia


In [ ]:
df.tail()

,Name,Brand channel,Subscribers (millions),Primary language,Category,Country
45,ARY Digital HD,Yes,59.7,Urdu,Entertainment,Pakistan
46,Fede Vigevani,No,59.4,UNKNOWN,Entertainment,Mexico
47,LooLoo Kids,UNKNOWN,59.3,English,Music,Romania
48,BillionSurpriseToys,Yes,UNKNOWN,English,Entertainment,United States
49,Marshmello,No,57.5,English,Music,United States


In [ ]:
df.isna().sum()

,0
Name,0
Brand channel,0
Subscribers (millions),0
Primary language,0
Category,0
Country,0


In [ ]:
df = df[['Name', 'Subscribers (millions)', 'Category', 'Country']]
df.head()

,Name,Subscribers (millions),Category,Country
0,MrBeast,335,Entertainment,United States
1,T-Series,280,Music,India
2,Cocomelon - Nursery Rhymes,UNKNOWN,Education,United States
3,SET India,UNKNOWN,Entertainment,India
4,Vlad and Niki,129,Entertainment,Russia


In [ ]:
df['SuSubscribers (millions)'] = pd.to_numeric(df['Subscribers (millions)'], errors='coerce')


In [ ]:
df.dtypes

,0
Name,object
Subscribers (millions),float64
Category,object
Country,object


In [ ]:
df = df.replace('UNKNOWN', None).dropna()
df.isnull().sum()

,0
Name,0
Subscribers (millions),0
Category,0
Country,0
SuSubscribers (millions),0


# Conclusion

In this acticity, we use the pandas library from python to transform those long data into clean data ready for analysis, like if need only the first 5 or the last 5, you just need to declare the df.head() and df.tail() and .isna() to find a missing data then display it on screen whihc of them has no value in the sheet data and also the .dropna() which is used for removing missing values in the set making the data, complete and reliable records.

Good job on finishing the activity! Kindly submit your accomplished google colab link and a PDF file of this activity.